## Post-deployment QA: Reply tool Instrumentation - New VEFU Events 

This task is about ensuring that the newly added events to the [VisualEditorFeatureUse](https://meta.wikimedia.org/wiki/Schema:VisualEditorFeatureUse) schema are being logged correctly in database.

Test scenarios and status are documented in this [QC document](https://docs.google.com/spreadsheets/d/1txtypYKZHCiZEFnkye9LX7riF3VEippMmvN6ILs_BUQ/edit#gid=605234531)

# Contents

1. [Advanced Affordance Events](#1.-Advanced-Affordance-Events)
2. [Dialog Prevent Show Event](#2.-Dialog-Prevent-Show-Event)
3. [Special Characters Toolbar Events](#3.-Special-Characters-Toolbar-Events)

    

In [3]:
shhh <- function(expr) suppressPackageStartupMessages(suppressWarnings(suppressMessages(expr)))
shhh({
    library(tidyverse); library(glue); library(lubridate); library(scales)
})

In [6]:
# Collect VEFU events after deployment
query <- 
"SELECT
    date_format(dt, 'yyyy-MM-dd') AS event_dt,
    wiki AS wiki,
    event.editor_interface AS editor_interface,
    event.platform AS platform,
    event.feature As feature,
    event.integration AS integration,
    event.action AS action,
    COUNT(*) AS events
FROM event_sanitized.visualeditorfeatureuse
WHERE 
-- reviewing a few days prior to confirm when events started firing
    year = 2020 
    AND month >= 09
    AND wiki <> 'testwiki'
GROUP BY
    date_format(dt, 'yyyy-MM-dd'),
    wiki,
    event.editor_interface,
    event.feature,
    event.integration,
    event.platform,
    event.action
"

In [7]:
collect_vefu_events <- wmfdata::query_hive(query)

Don't forget to authenticate with Kerberos using kinit



In [8]:
collect_vefu_events$event_dt <- as.Date(collect_vefu_events$event_dt, format = "%Y-%m-%d")

# 1. Advanced Affordance Events

Event added that logs when people "open" the Reply Tool's advanced "shelf". 

This functionality can be accessed by clicking the Advanced link that will appear beneath the text input area and can be used to customize the edit summary that accompanies the comments they post with the Reply Tool.

It was deployed on 17 September 2020 to all wikis where the reply tool was accessible. 

**Event**	**Name**
* An event that fires when the Advanced link/affordance is clicked to "show" the Comment summary field	feature: dtReply, action: advanced-show
* An event that fires when the Advanced link/affordance is clicked to "hide" the Comment summary field	feature: dtReply, action: advanced-hide


[Task](https://phabricator.wikimedia.org/T261816)

## Confirm Events Firing as Expected

In [6]:
advanced_show_events <- collect_vefu_events %>%
    filter(action == 'advanced-show') %>%
    arrange(date)

advanced_show_events 

date       wiki          editor_interface platform feature integration    
1  2020-09-17 arwiki        visualeditor     desktop  dtReply discussiontools
2  2020-09-17 hewiki        wikitext         desktop  dtReply discussiontools
3  2020-09-18 frwiki        wikitext         desktop  dtReply discussiontools
4  2020-09-18 kowiki        wikitext         desktop  dtReply discussiontools
5  2020-09-18 jawiki        wikitext         desktop  dtReply discussiontools
6  2020-09-18 zhwiki        wikitext         desktop  dtReply discussiontools
7  2020-09-19 enwiki        wikitext         desktop  dtReply discussiontools
8  2020-09-19 jawiki        wikitext         desktop  dtReply discussiontools
9  2020-09-20 huwiki        visualeditor     desktop  dtReply discussiontools
10 2020-09-20 huwiki        wikitext         desktop  dtReply discussiontools
11 2020-09-20 kowiki        wikitext         desktop  dtReply discussiontools
12 2020-09-21 frwiki        visualeditor     desktop  dtReply discussiontools
13 2020-09-21 arwiki        wikitext         desktop  dtReply discussiontools
14 2020-09-22 jawiki        wikitext         desktop  dtReply discussiontools
15 2020-09-23 nlwiki        visualeditor     desktop  dtReply discussiontools
16 2020-09-23 viwiki        visualeditor     desktop  dtReply discussiontools
17 2020-09-24 huwiki        wikitext         desktop  dtReply discussiontools
18 2020-09-24 cswiki        visualeditor     desktop  dtReply discussiontools
19 2020-09-25 nlwiki        wikitext         desktop  dtReply discussiontools
20 2020-09-25 arwiki        visualeditor     desktop  dtReply discussiontools
21 2020-09-25 arwiki        wikitext         desktop  dtReply discussiontools
22 2020-09-27 huwiki        wikitext         desktop  dtReply discussiontools
23 2020-09-27 svwiki        wikitext         desktop  dtReply discussiontools
24 2020-09-28 svwiki        wikitext         desktop  dtReply discussiontools
25 2020-09-29 frwiki        visualeditor     desktop  dtReply discussiontools
26 2020-09-29 svwiki        wikitext         desktop  dtReply discussiontools
27 2020-09-29 nlwiki        wikitext         desktop  dtReply discussiontools
28 2020-09-29 enwiki        visualeditor     desktop  dtReply discussiontools
29 2020-09-30 svwiki        wikitext         desktop  dtReply discussiontools
30 2020-10-01 svwiki        wikitext         desktop  dtReply discussiontools
⋮  ⋮          ⋮             ⋮                ⋮        ⋮       ⋮              
59 2020-10-15 arwiki        visualeditor     desktop  dtReply discussiontools
60 2020-10-15 enwikiversity wikitext         desktop  dtReply discussiontools
61 2020-10-15 metawiki      wikitext         desktop  dtReply discussiontools
62 2020-10-16 kowiki        wikitext         desktop  dtReply discussiontools
63 2020-10-16 svwiki        visualeditor     desktop  dtReply discussiontools
64 2020-10-16 arwiki        wikitext         desktop  dtReply discussiontools
65 2020-10-17 svwiki        wikitext         desktop  dtReply discussiontools
66 2020-10-17 nlwiki        wikitext         desktop  dtReply discussiontools
67 2020-10-18 nlwiki        wikitext         desktop  dtReply discussiontools
68 2020-10-18 huwiki        wikitext         desktop  dtReply discussiontools
69 2020-10-18 fawiki        wikitext         desktop  dtReply discussiontools
70 2020-10-18 svwiki        wikitext         desktop  dtReply discussiontools
71 2020-10-19 fawiki        visualeditor     desktop  dtReply discussiontools
72 2020-10-19 itwiki        wikitext         desktop  dtReply discussiontools
73 2020-10-19 nlwiki        wikitext         desktop  dtReply discussiontools
74 2020-10-19 ptwiki        wikitext         desktop  dtReply discussiontools
75 2020-10-20 nlwiki        wikitext         desktop  dtReply discussiontools
76 2020-10-21 nlwiki        wikitext         desktop  dtReply discussiontools
77 2020-10-22 nlwiki        visualeditor     desktop  dtReply discussiontools
78 2020-10-22 nlwiki     

Events are being recorded as early as 17 September 2020, which was the date of deployment. 

Confirmed that advanced show events are only being recorded for dtreply feature and discussiontools integration as expected. Events are recorded for both wikitext and visualeditor.

In [7]:
advanced_close_events <- collect_vefu_events %>%
    filter(action == 'advanced-hide') %>%
    arrange(date)

advanced_close_events 

date       wiki          editor_interface platform feature integration    
1  2020-09-17 arwiki        visualeditor     desktop  dtReply discussiontools
2  2020-09-17 hewiki        wikitext         desktop  dtReply discussiontools
3  2020-09-18 jawiki        wikitext         desktop  dtReply discussiontools
4  2020-09-18 zhwiki        wikitext         desktop  dtReply discussiontools
5  2020-09-19 jawiki        wikitext         desktop  dtReply discussiontools
6  2020-09-20 huwiki        wikitext         desktop  dtReply discussiontools
7  2020-09-21 arwiki        wikitext         desktop  dtReply discussiontools
8  2020-09-28 svwiki        wikitext         desktop  dtReply discussiontools
9  2020-09-29 enwiki        visualeditor     desktop  dtReply discussiontools
10 2020-09-29 frwiki        visualeditor     desktop  dtReply discussiontools
11 2020-09-29 svwiki        wikitext         desktop  dtReply discussiontools
12 2020-10-01 zhwiki        visualeditor     desktop  dtReply discussiontools
13 2020-10-02 cswiki        wikitext         desktop  dtReply discussiontools
14 2020-10-04 frwiki        wikitext         desktop  dtReply discussiontools
15 2020-10-05 arwiki        visualeditor     desktop  dtReply discussiontools
16 2020-10-05 nlwiki        wikitext         desktop  dtReply discussiontools
17 2020-10-07 frwiki        wikitext         desktop  dtReply discussiontools
18 2020-10-10 eswiki        wikitext         desktop  dtReply discussiontools
19 2020-10-10 frwiki        visualeditor     desktop  dtReply discussiontools
20 2020-10-12 arwiki        wikitext         desktop  dtReply discussiontools
21 2020-10-13 nlwiki        visualeditor     desktop  dtReply discussiontools
22 2020-10-14 kawiki        visualeditor     desktop  dtReply discussiontools
23 2020-10-15 enwikiversity wikitext         desktop  dtReply discussiontools
24 2020-10-16 arwiki        wikitext         desktop  dtReply discussiontools
25 2020-10-18 fawiki        wikitext         desktop  dtReply discussiontools
26 2020-10-19 ptwiki        wikitext         desktop  dtReply discussiontools
27 2020-10-19 fawiki        visualeditor     desktop  dtReply discussiontools
28 2020-10-19 itwiki        wikitext         desktop  dtReply discussiontools
29 2020-10-22 nlwiki        visualeditor     desktop  dtReply discussiontools
30 2020-10-22 nlwiki        wikitext         desktop  dtReply discussiontools
31 2020-10-22 arwiki        wikitext         desktop  dtReply discussiontools
32 2020-10-23 arwiki        visualeditor     desktop  dtReply discussiontools
33 2020-10-23 fawiki        wikitext         desktop  dtReply discussiontools
34 2020-10-24 plwiki        wikitext         desktop  dtReply discussiontools
35 2020-10-26 cswiki        visualeditor     desktop  dtReply discussiontools
   action        events
1  advanced-hide  1    
2  advanced-hide  1    
3  advanced-hide  1    
4  advanced-hide  1    
5  advanced-hide  1    
6  advanced-hide  2    
7  advanced-hide  1    
8  advanced-hide  5    
9  advanced-hide  6    
10 advanced-hide  9    
11 advanced-hide 12    
12 advanced-hide  1    
13 advanced-hide  1    
14 advanced-hide  1    
15 advanced-hide  1    
16 advanced-hide  1    
17 advanced-hide  2    
18 advanced-hide  1    
19 advanced-hide  1    
20 advanced-hide  1    
21 advanced-hide  1    
22 advanced-hide  1    
23 advanced-hide  1    
24 advanced-hide  1    
25 advanced-hide  4    
26 advanced-hide  1    
27 advanced-hide  1    
28 advanced-hide  1    
29 advanced-hide  1    
30 advanced-hide  3    
31 advanced-hide  1    
32 advanced-hide  1    
33 advanced-hide  2    
34 advanced-hide  1    
35 advanced-hide  1

Confirmed there are fewer advanced hide events compared to advanced show events as expected. Advanced hide events also start coming in on 17 September 2020 (date of deployment).

Confirmed that advanced hide events are only being recorded for dtreply feature and discussiontools integration as expected. Events are recorded for both wikitext and visualeditor.

## Total Events

Review total events to confirm there are fewer advanced-hide events to advanced-show events.

In [8]:
total_advanced_events <- collect_vefu_events %>%
    filter(action == 'advanced-hide' |
          action == 'advanced-show') %>%
    group_by(action)  %>%
    summarise(total_events = sum(events))

total_advanced_events 

`summarise()` ungrouping output (override with `.groups` argument)



action        total_events
1 advanced-hide  71         
2 advanced-show 142

Confirmed that there are more advanced show events compared to advanced hide. As of 26 Oct 2020, there have been 142 advanced-show events and 71 advanced-hide events (213 advanced affordance events total)

There are no events in any other platform other than "Desktop" since this feature is only available on desktop.

## Event Counts by Wiki

In [9]:
total_advanced_events_bywiki <- collect_vefu_events %>%
    filter(action == 'advanced-hide' |
          action == 'advanced-show') %>%
    group_by(action, wiki)  %>%
    summarise(total_events = sum(events)) %>%
    arrange(wiki)

total_advanced_events_bywiki

`summarise()` regrouping output by 'action' (override with `.groups` argument)



action        wiki          total_events
1  advanced-hide arwiki         7          
2  advanced-show arwiki        14          
3  advanced-hide cswiki         2          
4  advanced-show cswiki         3          
5  advanced-hide enwiki         6          
6  advanced-show enwiki         7          
7  advanced-hide enwikiversity  1          
8  advanced-show enwikiversity  1          
9  advanced-hide eswiki         1          
10 advanced-show eswiki         2          
11 advanced-hide fawiki         7          
12 advanced-show fawiki         8          
13 advanced-hide frwiki        13          
14 advanced-show frwiki        15          
15 advanced-hide hewiki         1          
16 advanced-show hewiki         1          
17 advanced-hide huwiki         2          
18 advanced-show huwiki        11          
19 advanced-hide itwiki         1          
20 advanced-show itwiki         1          
21 advanced-hide jawiki         2          
22 advanced-show jawiki         3          
23 advanced-hide kawiki         1          
24 advanced-show kawiki         1          
25 advanced-show kowiki         5          
26 advanced-show metawiki       1          
27 advanced-hide nlwiki         6          
28 advanced-show nlwiki        26          
29 advanced-hide plwiki         1          
30 advanced-show plwiki         1          
31 advanced-hide ptwiki         1          
32 advanced-show ptwiki         1          
33 advanced-hide svwiki        17          
34 advanced-show svwiki        38          
35 advanced-show viwiki         1          
36 advanced-hide zhwiki         2          
37 advanced-show zhwiki         2

Advanced events recorded for arwiki, cswiki, enwiki, frwiki, hewiki, huwiki, jawiki, kowiki, nlwiki, svwiki, viwiki, and zhwiki. A few observations:
* For all wikis, there are more advanced-show events then advanced-hide events as expected.
* For viwiki and kowiki, there is only one advanced-show event recorded (no advanced-hide event).
* Enwiki and HeWiki have events but are not listed on the deployment wikis; however, these are likely wikis where it was enabled for testing using ?dtenable=1.


In [10]:
total_advanced_events_bywiki_both <- collect_vefu_events %>%
    filter(action == 'advanced-hide' |
          action == 'advanced-show') %>%
    group_by( wiki)  %>%
    summarise(total_events = sum(events)) %>%
    arrange(total_events)

total_advanced_events_bywiki_both 

`summarise()` ungrouping output (override with `.groups` argument)



wiki          total_events
1  metawiki       1          
2  viwiki         1          
3  enwikiversity  2          
4  hewiki         2          
5  itwiki         2          
6  kawiki         2          
7  plwiki         2          
8  ptwiki         2          
9  eswiki         3          
10 zhwiki         4          
11 cswiki         5          
12 jawiki         5          
13 kowiki         5          
14 enwiki        13          
15 huwiki        13          
16 fawiki        15          
17 arwiki        21          
18 frwiki        28          
19 nlwiki        32          
20 svwiki        55

Svwiki and Nlwiki have the largest number of advanced affordance events recorded to date (55 for svwiki and 32 for nlwiki).

# 2. Dialog Prevent Show Event 
Event added that is fired when someone attempts to switch from the Reply Tool's source to visual mode after drafting a comment that contains any combination of the following syntax: extension, template or table.

This is associated with the editor_switch feature and discussiontools integration.

[Task](https://phabricator.wikimedia.org/T259673)

## Confirm events firing as expected

In [44]:
switch_prevent_events <- collect_vefu_events %>%
    filter(action == 'dialog-prevent-show') %>%
    arrange(desc(date))

switch_prevent_events

date       wiki   editor_interface platform feature       integration    
1 2020-11-10 nlwiki wikitext         desktop  editor-switch discussiontools
2 2020-11-09 nlwiki visualeditor     desktop  editor-switch discussiontools
3 2020-11-03 arwiki visualeditor     desktop  editor-switch discussiontools
4 2020-10-30 plwiki visualeditor     desktop  editor-switch discussiontools
5 2020-10-25 arwiki visualeditor     desktop  editor-switch discussiontools
6 2020-10-16 arwiki visualeditor     desktop  editor-switch discussiontools
7 2020-10-14 frwiki visualeditor     desktop  editor-switch discussiontools
8 2020-10-13 nlwiki visualeditor     desktop  editor-switch discussiontools
  action              events
1 dialog-prevent-show 1     
2 dialog-prevent-show 2     
3 dialog-prevent-show 1     
4 dialog-prevent-show 1     
5 dialog-prevent-show 1     
6 dialog-prevent-show 1     
7 dialog-prevent-show 1     
8 dialog-prevent-show 2

Since the deployment date, only 7 dialog-prevent-show events have been recorded. Many of these are likely testing events. It appears that this dialog appears very rarely based on the current data. 

All events are for the editor_switch feature and discussion tools integration as expected. The editing interface for these events are recorded as 'visual editor' - Is that correct given that this action is supposed to record when an editor is prevented from switching to visual mode from source?

Logged an event on Fr.Wiki, cswiki, and kowiki. May take some time for event to come in. 

# 3. Special Characters Toolbar Events

Events added to collect data for when people are using the visual editor and wikieditor special characters toolbar. Note: Support for this feature has not yet been added for the reply tool. This instrumentation was added to inform how frequently users are using the special character's toolbar in other editing interfaces.

New events deployed on 8 October 2020
[Task](https://phabricator.wikimedia.org/T264146)

## Events by date

In [11]:
special_character_events_bydate <- collect_vefu_events %>%
    filter(feature == 'specialCharacter') %>%
    arrange(event_dt)

head(special_character_events_bydate)

event_dt   wiki          editor_interface platform feature         
1 2020-10-13 mediawikiwiki visualeditor     desktop  specialCharacter
2 2020-10-13 mediawikiwiki visualeditor     desktop  specialCharacter
3 2020-10-13 mediawikiwiki visualeditor     desktop  specialCharacter
4 2020-10-14 mediawikiwiki visualeditor     desktop  specialCharacter
5 2020-10-14 mediawikiwiki visualeditor     desktop  specialCharacter
6 2020-10-15 enwiki        visualeditor     desktop  specialCharacter
  integration action                events
1 page        dialog-abort          1     
2 page        insert-latin          1     
3 page        window-open-from-tool 1     
4 page        window-open-from-tool 1     
5 page        dialog-abort          1     
6 page        dialog-abort          1

We first start recording events on 13 October 2020 as expected

## Events by action

In [14]:
special_character_events_byaction <- collect_vefu_events %>%
    filter(feature == 'specialCharacter') %>%
    group_by(action)  %>%
    summarise(num_events = sum(events)) %>%
    arrange(desc(num_events))

special_character_events_byaction

`summarise()` ungrouping output (override with `.groups` argument)



action                    num_events
1  window-open-from-tool     6744      
2  dialog-abort              5887      
3  insert-other              4994      
4  insert-latin              1646      
5  insert-symbols            1267      
6  insert-cyrillic            817      
7  insert-ipa                 699      
8  insert-bangla              405      
9  insert-greek               402      
10 insert-arabic              390      
11 insert-greekextended       117      
12 insert-arabicextended      107      
13 insert-hebrew              100      
14 insert-latinextended        84      
15 insert-canadianaboriginal   30      
16 insert-lao                  28      
17 insert-devanagari           23      
18 insert-telugu               21      
19 insert-thai                 17      
20 insert-khmer                11      
21 insert-gujarati              5      
22 insert-tamil                 3      
23 insert-sinhala               2

'Dialog-abort' and 'window-open-tool' are the most frequently recorded SpecialCharacter toolbar events. There are more events opening the special characters toolbar vs closing the toolbar as expected.

There are 21 different insert actions for when someone adds a character from the toolbar. `insert-other`, `insert-latin`, and `insert-symbols` are the most frequently used options. `insert-gujarati`, `insert-tamil`, and	`insert-sinhala`	are the least used actions. 



## Events by Platform, Integration and Editor Interface

In [15]:
special_character_events_bytype <- collect_vefu_events %>%
    filter(feature == 'specialCharacter') %>%
    group_by(platform, integration, editor_interface)  %>%
    summarise(num_events = sum(events)) %>%
    arrange(desc(num_events))

special_character_events_bytype

`summarise()` regrouping output by 'platform', 'integration' (override with `.groups` argument)



platform integration editor_interface num_events
1 desktop  page        visualeditor     20924     
2 desktop  page        wikitext-2017     2875

All special character events recorded since the instrumentation was deployed in October 2020 have been on desktop using both the visualeditor (VE in visual mode) or wikitext-2017 (VE in source mode) as expected. 

All special character events are recorded as main edit type events (`event.integration = 'page')`. This is also expected behavior as this instrumentation has not been deployed to the discussion tools. 



## Events by Wiki

In [19]:
special_character_events_bywiki <- collect_vefu_events %>%
    filter(feature == 'specialCharacter') %>%
    group_by(wiki)  %>%
    summarise(num_events = sum(events)) %>%
    arrange(desc(num_events))

head(special_character_events_bywiki)

`summarise()` ungrouping output (override with `.groups` argument)



wiki   num_events
1 ruwiki 3261      
2 enwiki 3191      
3 plwiki 2566      
4 frwiki 2030      
5 dewiki 1456      
6 eswiki 1304

In [18]:
n_distinct(special_character_events_bywiki$wiki)

[1] 122

We've recorded special character toolbar events on 122 different wikis. The most events were recorded for Russian Wikipedia, English Wikipedia, Polish Wikipedia, French Wikipedia, and German Wikipedia. 

Below is breakdown of the special character events by action for each of these wikis.

In [35]:
## Review the number of special character events by action for each of the top 6 wikis with the most recorded overall events
special_character_events_bywikiaction <- collect_vefu_events %>%
    filter(feature == 'specialCharacter',
            wiki %in% c('ruwiki', 'enwiki', 'plwiki', 'frwiki', 'dewiki', 'eswiki')) %>%
    group_by(wiki, action)  %>%
    summarise(num_events = sum(events)) %>%
    arrange(desc(wiki, num_events))

special_character_events_bywikiaction

`summarise()` regrouping output by 'wiki' (override with `.groups` argument)



wiki   action                    num_events
1  ruwiki dialog-abort              873       
2  ruwiki insert-arabic              14       
3  ruwiki insert-arabicextended       1       
4  ruwiki insert-bangla               2       
5  ruwiki insert-canadianaboriginal  13       
6  ruwiki insert-cyrillic            30       
7  ruwiki insert-greek               34       
8  ruwiki insert-greekextended       13       
9  ruwiki insert-hebrew              61       
10 ruwiki insert-ipa                 60       
11 ruwiki insert-lao                 10       
12 ruwiki insert-latin               52       
13 ruwiki insert-latinextended        2       
14 ruwiki insert-other              951       
15 ruwiki insert-symbols            197       
16 ruwiki insert-thai                 1       
17 ruwiki window-open-from-tool     947       
18 plwiki dialog-abort              219       
19 plwiki insert-arabic              85       
20 plwiki insert-arabicextended      35       
21 plwiki insert-bangla             338       
22 plwiki insert-canadianaboriginal   2       
23 plwiki insert-cyrillic           547       
24 plwiki insert-devanagari           1       
25 plwiki insert-greek               72       
26 plwiki insert-greekextended       94       
27 plwiki insert-gujarati             3       
28 plwiki insert-hebrew               3       
29 plwiki insert-ipa                 36       
30 plwiki insert-lao                 13       
⋮  ⋮      ⋮                         ⋮         
59 eswiki insert-ipa                 215      
60 eswiki insert-khmer                 1      
61 eswiki insert-latin               153      
62 eswiki insert-other               349      
63 eswiki insert-symbols              18      
64 eswiki insert-tamil                 1      
65 eswiki window-open-from-tool      291      
66 enwiki dialog-abort               939      
67 enwiki insert-arabic               15      
68 enwiki insert-bangla                1      
69 enwiki insert-canadianaboriginal    2      
70 enwiki insert-cyrillic             13      
71 enwiki insert-greek               103      
72 enwiki insert-greekextended         7      
73 enwiki insert-hebrew               25      
74 enwiki insert-ipa                  47      
75 enwiki insert-khmer                 1      
76 enwiki insert-latin               241      
77 enwiki insert-latinextended         5      
78 enwiki insert-other               600      
79 enwiki insert-sinhala               1      
80 enwiki insert-symbols             125      
81 enwiki window-open-from-tool     1066      
82 dewiki dialog-abort               401      
83 dewiki insert-greek                 1      
84 dewiki insert-latin                39      
85 dewiki insert-latinextended         2      
86 dewiki insert-other               525      
87 dewiki insert-symbols              18      
88 dewiki window-open-from-tool      470

## Check to make sure specialCharacter specific features are not recorded elsewhere

In [26]:
insert_type_events <- collect_vefu_events %>%
    filter(grepl('^insert', action)) %>%
    group_by(feature)  %>%
    summarise(num_events = sum(events)) %>%
    arrange(desc(num_events))

insert_type_events

`summarise()` ungrouping output (override with `.groups` argument)



feature          num_events
1 table            104404    
2 specialCharacter  11168

Insert type actions are recorded only for specialCharacter and table features as expected.  